bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManager.java

```diff
@@ -508,29 +509,130 @@ public Void call() throws Exception {
         }
     }
 
+    private void initializeHandler(Thing thing) {
+        if (isInitializable(thing)) {
+            ThingStatusInfo statusInfo = buildStatusInfo(ThingStatus.INITIALIZING, ThingStatusDetail.NONE);
+            setThingStatus(thing, statusInfo);
+            initializeHandler(thing, thing.getHandler());
+        } else {
+            logger.debug("Thing '{}' not initializeable, check required configuration parameters.", thing.getUID());
+            ThingStatusInfo statusInfo = buildStatusInfo(ThingStatus.UNINITIALIZED,
+                    ThingStatusDetail.HANDLER_CONFIGURATION_PENDING);
+            setThingStatus(thing, statusInfo);
+        }
+    }
+
+    private boolean isInitializable(Thing thing) {
+        // determines if all 'required' configuration parameters are available in the configuration
+        logger.debug("Determine if thing '{}' is initializable.", thing.getUID());
+
+        ThingType thingType = TypeResolver.resolve(thing.getThingTypeUID());
+        if (thingType != null) {
+            ConfigDescription description = TypeResolver.resolve(thingType.getConfigDescriptionURI());
+            if (description == null) {
+                return true;
+            }
+
+            List<String> requiredParameters = getRequiredParameters(description);
+            Map<String, Object> properties = thing.getConfiguration().getProperties();
+            return properties.keySet().containsAll(requiredParameters);
+        } else {
+            logger.debug("Thing type '{}' of thing '{}' not found.", thing.getThingTypeUID(), thing.getUID());
+        }
+        return false;
+    }
+
+    private List<String> getRequiredParameters(ConfigDescription description) {
+        List<String> requiredParameters = new ArrayList<>();
+        for (ConfigDescriptionParameter param : description.getParameters()) {
+            if (param.isRequired()) {
+                requiredParameters.add(param.getName());
+            }
+        }
+        return requiredParameters;
+    }
+
+    private void initializeHandler(final Thing thing, final ThingHandler thingHandler) {
+        logger.debug("Calling initialize handler for thing '{}' at '{}'.", thing.getUID(), thingHandler);
+        try {
+            SafeMethodCaller.call(new SafeMethodCaller.ActionWithException<Void>() {
+                @Override
+                public Void call() throws Exception {
+                    thingHandler.initialize();
+                    return null;
+                }
+            });
+        } catch (TimeoutException ex) {
+            ThingStatusInfo statusInfo = buildStatusInfo(ThingStatus.UNINITIALIZED,
+                    ThingStatusDetail.HANDLER_INITIALIZING_ERROR,
+                    ex.getCause() != null ? ex.getCause().getMessage() : ex.getMessage());
+            setThingStatus(thing, statusInfo);
+            logger.warn("Initializing handler for thing '{}' takes more than {}ms.", thing.getUID(),
```

Comment:
Do we need this since we already have a log by the thing status update with the same message?